src/bytes.rs

```diff
@@ -51,28 +50,39 @@ impl<const BITS: usize, const LIMBS: usize> Uint<BITS, LIMBS> {
     /// Access the underlying store as a little-endian bytes.
     ///
     /// Uses an optimized implementation on little-endian targets.
+    #[cfg(feature = "alloc")]
     #[must_use]
-    #[inline(always)]
+    #[inline]
     pub const fn as_le_bytes(&self) -> Cow<'_, [u8]> {
         // On little endian platforms this is a no-op.
         #[cfg(target_endian = "little")]
         return Cow::Borrowed(self.as_le_slice());
 
-        // In others it's a bit more complicated.
-        #[cfg(not(target_endian = "little"))]
-        return Cow::Owned(self.to_le_bytes_vec());
+        // In others, reverse each limb and return a copy.
+        #[cfg(target_endian = "big")]
+        return Cow::Owned({
+            let mut cpy = *self;
+            cpy.limbs.iter_mut().for_each(|limb| limb.reverse_bits());
+            slice::from_raw_parts(cpy.limbs.as_ptr().cast(), Self::BYTES).to_vec()
+        });
```

Comment:
This used to recurse infinitely (`to_le_bytes_vec` calls `as_le_bytes().into_owned()` which is this exact expression), make it actually do something on BE targets